src/include/btree.i

```diff
@@ -970,7 +970,7 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_PAGE *page, int check_splits)
 	 * a transaction value, once that's globally visible, we know we can
 	 * evict the created page.
 	 */
-	if (WT_PAGE_IS_INTERNAL(page) &&
+	if (check_splits && WT_PAGE_IS_INTERNAL(page) &&
```

Comment:
This part of the change deserves some explanation and thought.  I don't know for sure that this is necessary, but I'm now asserting that all pages in a clean tree can be safely discarded, and this is the one case I'm not sure is guaranteed anywhere.

We could include `mod->mod_split_txn` in the calculation of `btree->rec_max_txn`, maybe?